ejb/WFLY-4699-expose-ejb3-deployment-info.adoc

```diff
@@ -0,0 +1,516 @@
+                "month" => "*",
+                "day-of-month" => "*",
+                "day-of-week" => "*",
+                "hour" => "0",
+                "minute" => "0",
+                "second" => "15",
+                "timezone" => undefined,
+                "start" => undefined,
+                "end" => undefined
+            }
+        }],
+        "transaction-type" => "CONTAINER",
+        "wait-time" => 0L,
+        "service" => {"timer-service" => undefined}
+    }
+}
+
+[source]
+/deployment=ejb-management.jar/subsystem=ejb3/message-driven-bean=NoTimerMDB:read-resource(include-runtime)
+{
+    "outcome" => "success",
+    "result" => {
+        "activation-config" => [
+            ("destination" => "java:/queue/NoTimerMDB-queue"),
+            ("destinationType" => "javax.jms.Queue"),
+            ("acknowledgeMode" => "Auto-acknowledge")
+        ],
+        "component-class-name" => "sample.NoTimerMDB",
+        "declared-roles" => [
+            "Role3",
+            "Role2",
+            "Role1"
+        ],
+        "delivery-active" => true,
+        "execution-time" => 0L,
+        "invocations" => 0L,
+        "message-destination-link" => "queue/NoTimerMDB-queue",
+        "message-destination-type" => "javax.jms.Queue",
+        "messaging-type" => "javax.jms.MessageListener",
+        "methods" => {},
+        "peak-concurrent-invocations" => 0L,
+        "pool-available-count" => 16,
+        "pool-create-count" => 0,
+        "pool-current-size" => 0,
+        "pool-max-size" => 16,
+        "pool-name" => "mdb-strict-max-pool",
+        "pool-remove-count" => 0,
+        "run-as-role" => "Role3",
+        "security-domain" => "other",
+        "timeout-method" => undefined,
+        "timers" => [],
+        "transaction-type" => "CONTAINER",
+        "wait-time" => 0L,
+        "service" => undefined
+    }
+}
+
+=== Nice-to-Have Requirements
+
+* ability to read these deployment information from admin console
+
+=== Non-Requirements
+
+* ability to modify these deployment information via CLI or admin console;
+* ability to expose deployment information from jboss-specific deployment descriptors;
+* ability to expose ALL deployment information from ejb-jar.xml and annotations for an EJB.
+* ability to expose deployment information from sources other than ejb3.
+
+== Test Plan
+
+New tests will be added to `testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/management/deployments`.
+
+* verify that deployment information should be available as runtime info for all ejb bean types:
+** stateless
+** stateful
+** singleton
+** MDB
+* verify that absence of certain optional deployment information will not cause errors (e.g., NullPointerException,
+IllegalArgumentException, etc) when reading this bean's runtime management resources.
+* verify the correctly merged value is exposed as runtime resources via CLI, when some deployment information is specified in both ejb-jar.xml and annotations.
+
```

Comment:
@chengfang what about also testing that information can be accessed in secured EJB? is there any security restriction that applies to cli?

Reply:
I think this feature is not related to security, and there is no extra security restriction that applies to cli, other than the existing cli user/password authentication.